balboa-agent/src/main/scala/com/socrata/balboa/agent/metrics/BalboaAgentMetrics.scala

```diff
@@ -83,17 +83,23 @@ object BalboaAgentMetrics {
   val errorInvalidValueCounter: Counter = registry.counter(MetricRegistry.name(name, "error", "invalid-value"))
 
   /**
-    * Creates a metric checking the size of a single directory.
+    * Creates a metric that counts a directory for files within that directory
     *
-    * @param metricName The name of the metric to register.
-    * @param directory The directory to check.
-    * @return Success[Metric] if the metric was created.  Failure[IllegalArgumentException] if it wasn't able to use the defined directory.
+    * @param metricName The name of the metric to create.
+    * @param directory The directory to count metrics in.
+    * @param filter Filter to use to count files.  If none returns all files
+    * @return Success(Metric) if successfully create the metric, False otherwise
     */
-  def directorySize(metricName: String, directory: File): Try[Metric] = directory match {
+  def numFiles(metricName: String, directory: File, filter: Option[FileFilter]): Try[Metric] = directory match {
     case f: File if directory.exists() && directory.isDirectory =>
-      Success(registry.register(MetricRegistry.name(name, metricName, "size"),
+      Success(registry.register(MetricRegistry.name(name, metricName, "num", "files"),
         new CachedGauge[Int](5, TimeUnit.MINUTES) { //
-        override def loadValue(): Int = FileUtils.getDirectories(directory).size()
+        override def loadValue(): Int = FileUtils.getDirectories(directory).map(dir =>
```

Comment:
I am a little worried about FIle IO race conditions that could cause unexpected exceptions.  Right now we are assuming the directory does not get deleted prior to calling list files.